src/steamship/client/steamship.py

```diff
@@ -88,6 +90,32 @@ def embed_and_search(
             expect=QueryResults,
         )
 
+    @staticmethod
+    @contextmanager
+    def temporary_workspace(**kwargs) -> Generator["Steamship", None, None]:
+        """Create a client rooted in a temporary workspace that will be deleted after use."""
+        # Create a new client and switch to a temporary workspace
+        client = Steamship(**kwargs)
+        temporary_handle = "temp-" + str(uuid.uuid4())
+        client.switch_workspace(temporary_handle)
+
+        # Safety check that we are now working form the new workspace.
+        if client.config.workspace_handle != temporary_handle:
+            raise SteamshipError(
+                message=f"Attempted to switch totemporary workspace {temporary_handle} but the client claimed to be working from {client.config.workspace_handle}"
+            )
+
+        yield client
+
```

Comment:
Would it be better to store a different handle to the client, so that if the user switched workspaces on it, it could still be deleted?